core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -1439,7 +1439,7 @@ protected String getDateTimeFormatString(
         startIndex = i + 1;
       }
     }
-    if (lastIndex > startIndex) {
+    if (lastIndex >= startIndex) {
```

Comment:
checking for gt and eq will result in test failure for some scenarios. Kindly retest for all sscenarios